### PR TITLE
Adding an automated test for curator notification

### LINKED ISSUE #359: Notify the collection moderators when a item in the collection is marked complete
This can be accomplished via the after on the AASM,  like [publish](https://github.com/pulibrary/pdc_describe/blob/main/app/models/work.rb#L35)

This is complete when

- [x] The moderator(s) of a collection receives an in app notification when a dataset is marked as complete
- [ ] There is an automated test that checks that these notifications get sent

### PR DESCRIPTION
**Pull request by carolyncole**
fixes #359

### CODE REVIEW

File: spec/system/form_submission_spec.rb [hunk 1]
```diff
@@ -75,6 +77,15 @@
...
       expect(page).to have_content "awaiting_approval"
+
+      # Now sign is as the collection curator and see the notification on your dashboard
+      sign_out user
+      sign_in curator
+      visit(user_path(curator))
+      expect(page).to have_content curator.display_name
+      within("#unfinished_datasets span.badge.rounded-pill.bg-primary") do
+        expect(page).to have_content "1"
+      end
```
Comment (mccalluc) [inline spec/system/form_submission_spec.rb:88]:
```suggestion
      within("#unfinished_datasets") do
        expect(page).to have_content "Sowing the Seeds for More Usable Web Archives: A Usability Study of Archive-It"
        expect(page).to have_content "awaiting_approval"
      end
```
The exact element path doesn't seem like a requirement, but seeing the title and the status does.

**Review by mccalluc — APPROVED** [review]
Looks good: A couple suggestions.

File: spec/system/form_submission_spec.rb (hunk 1 above)
Comment (carolyncole) [reply spec/system/form_submission_spec.rb:88]: @mccalluc I'm looking for the badge with the 1 here.  I'll add a comment

File: spec/system/form_submission_spec.rb [hunk 2]
```diff
@@ -39,6 +40,7 @@
   end
   context "happy path" do
     it "produces and saves a valid datacite record", js: true do
+      curator # make sure the curator exists prior to starting this process so they can be notified
```
[the remark this answers is not in the document]
Comment (carolyncole) [reply spec/system/form_submission_spec.rb:43]: This was just to construct the item, I could have done a let! instead.  Maybe that would be better

**Comment by carolyncole** [comment]
@mccalluc do these changes make more sense to you?

**Review by mccalluc — APPROVED** [review]

**Pull request merged**